k2/csrc/array_ops.h

```diff
@@ -198,6 +198,15 @@ T MaxValue(const Array1<T> &src) {
   return MaxValue(src.Context(), src.Dim(), src.Data());
 }
 
+template <typename T>
+T Sum(const Array1<T> &src) {
```

Comment:
The function name `Sum` is a little confused, as `ExclusiveSum` does not count the last element in `src`.

Reply:
Will fix.

Reply:
Fixed in #623

Reply:
mm, I don't think we need the function to sum all values in Array, I just feel what we want is the last element of `exclusive_sum` as @danpovey implemented, I commented as I think the name `Sum` may be confusing here.

Reply:
You don't need to sum all values. You can specify a range to sum.

Reply:
No, I meant all values.  The user could specify a range by calling it with a sub-range of the array, if needed.